From 5eea70b9d5852e468467c1565927dbe0c76d8674 Mon Sep 17 00:00:00 2001
From: gaurank kathpalia <gkathpal@codeaurora.org>
Date: Mon, 23 Apr 2018 08:07:46 +0530
Subject: qcacld-3.0: Remove cds_pkt_free in assoc request api

In the API lim_send_assoc_req_mgmt_frame, the host
allocates memory for the assoc request packet
taking all inputs of payload and the mac header
size etc, and in case the mem allocation fails
it clears away the memory allocated to the packet
with cds packet free, which was not even allocated

Fix is to remove the packet free in case of memory not
allocated

Change-Id: I3fb75b1947dfe039605c42aa19c2d0bacc7bf55d
CRs-Fixed: 2216741
---
 core/mac/src/pe/lim/lim_send_management_frames.c | 2 --
 1 file changed, 2 deletions(-)

diff --git a/core/mac/src/pe/lim/lim_send_management_frames.c b/core/mac/src/pe/lim/lim_send_management_frames.c
index 3dcfbc7..5f21a86 100644
--- a/core/mac/src/pe/lim/lim_send_management_frames.c
+++ b/core/mac/src/pe/lim/lim_send_management_frames.c
@@ -1956,8 +1956,6 @@ lim_send_assoc_req_mgmt_frame(tpAniSirGlobal mac_ctx,
 
 		assoc_cnf.resultCode = eSIR_SME_RESOURCES_UNAVAILABLE;
 
-		cds_packet_free((void *)packet);
-
 		lim_post_sme_message(mac_ctx, LIM_MLM_ASSOC_CNF,
 			(uint32_t *) &assoc_cnf);
 
-- 
cgit v1.1

